tests/lib/rules/no-potential-component-option-typo.js

```diff
@@ -0,0 +1,406 @@
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-potential-component-option-typo')
+
+const RuleTester = require('eslint').RuleTester
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const tester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: { ecmaVersion: 2018, sourceType: 'module' }
+})
+const path = require('path')
+const fs = require('fs')
+const VueComponentJsonCheckVuePath = path.resolve(
+  __dirname,
+  '../../fixtures/no-potential-component-option-typo/index.vue'
+)
+const VueComponentJsonCheckVue = fs.readFileSync(
+  VueComponentJsonCheckVuePath,
+  'utf8'
+)
+
+tester.run('no-potential-component-option-typo', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        dat: {},
+        method: {}
+      }
+      </script>
+      `,
+      // because vue preset is include by default, set the presets to empty
+      options: [{ presets: [] }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+      }
+      </script>
+      `,
+      options: [{ presets: ['vue'] }]
+    },
+    // test if give preset and the potentialTypoList length is zero, just for 100% test cover
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {}
+      }
+      </script>
+      `,
+      options: [{ presets: ['vue'] }]
+    },
+    // multi preset that won't report
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {},
+        beforeRouteEnter() {}
+      }
+      </script>
+      `,
+      options: [{ presets: ['vue', 'vue-router'] }]
+    },
+    //  test custom option that is not available in the presets
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        custom: {},
+        foo: {}
+      }
+      </script>
+      `,
+      options: [{ custom: ['custom', 'foo'] }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        abcde: {},
+        abcd: {}
+      }
+      </script>
+      `,
+      options: [{ custom: ['abcde', 'abcd'] }]
+    },
+    // valid test case set custom and threshold
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        custom: {},
+        fooooo: {}
+      }
+      </script>
+      `,
+      options: [{ custom: ['custom', 'foo'], threshold: 2 }]
+    },
+    // test all valid vue options
+    {
+      filename: 'test.vue',
+      code: VueComponentJsonCheckVue,
```

Comment:
Hmm.. I don't think it makes sense to put it in a separate file.
Could you put the code inline here?